### PR TITLE
Callback support for models

### PR DESCRIPTION
**Pull request by Michsior14**
This commit adds callbacks like AR has:

before_save
before_create/before_update/before_delete
INSERT/UPDATE/DELETE IN DB
after_create/after_update/after_delete
after_save

### CODE REVIEW

File: callbacks.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"reflect"
+
+	"github.com/pkg/errors"
+)
+
+func (m *Model) runCallbacks(c *Connection, name string) error {
+	rv := reflect.ValueOf(m.Value)
+	mv := rv.MethodByName(name)
+	if mv.IsValid() {
+		if mv.Type().NumOut() != 1 {
```
Comment (markbates) [inline callbacks.go]: You should check the type of the argument as well as the length.

File: executors.go [hunk 2]
```diff
@@ -36,14 +36,26 @@ func (c *Connection) ValidateAndSave(model interface{}, excludeColumns ...string
 
 var emptyUUID = uuid.Nil.String()
 
-func (c *Connection) Save(model interface{}, excludeColumns ...string) error {
+func (c *Connection) Save(model interface{}, excludeColumns ...string) (err error) {
```
Comment (markbates) [inline executors.go]: Please remove the named return arguments. Thanks.

**Comment by markbates** [comment]
There is also a conflict in `executors.go` that needs to be resolved.

**Comment by Michsior14** [comment]
@markbates 
Looks like internal travis error:
[mysql] 2017/04/19 20:46:25 packets.go:33: read tcp 127.0.0.1:52756->127.0.0.1:3306: i/o timeout

**Comment by markbates** [comment]
Thanks! When you get a chance it would be great to have some documentation around this. Maybe event a flag to the model generator to generate stub methods for the callbacks.

**Pull request merged**